x/trigger/types/keys.go

```diff
@@ -0,0 +1,150 @@
+
+	// QuerierRoute defines the module's query routing key
+	QuerierRoute = ModuleName
+
+	TriggerIDLength  = 8
+	QueueIndexLength = 8
+	GasLimitLength   = 8
+)
+
+var (
+	TriggerKeyPrefix       = []byte{0x01}
+	EventListenerKeyPrefix = []byte{0x02}
+	QueueKeyPrefix         = []byte{0x03}
+	GasLimitKeyPrefix      = []byte{0x04}
+	NextTriggerIDKey       = []byte{0x05}
+	QueueStartIndexKey     = []byte{0x06}
+	QueueLengthKey         = []byte{0x07}
+)
+
+// GetEventListenerKey converts an event name and trigger ID into an event registry key format.
+func GetEventListenerKey(eventName string, id TriggerID) []byte {
+	triggerIDBytes := make([]byte, TriggerIDLength)
+	binary.BigEndian.PutUint64(triggerIDBytes, id)
+
+	key := GetEventListenerPrefix(eventName)
+	key = append(key, triggerIDBytes...)
+	return key
+}
+
+// GetEventListenerPrefix converts an event name into a prefix for event registry.
+func GetEventListenerPrefix(eventName string) []byte {
+	eventNameBytes := GetEventNameBytes(eventName)
+
+	key := EventListenerKeyPrefix
+	key = append(key, eventNameBytes...)
+	return key
+}
+
+// GetTriggerKey converts a trigger into key format.
+func GetTriggerKey(id TriggerID) []byte {
+	triggerIDBytes := make([]byte, TriggerIDLength)
+	binary.BigEndian.PutUint64(triggerIDBytes, id)
+
+	key := TriggerKeyPrefix
+	key = append(key, triggerIDBytes...)
+	return key
+}
+
+// GetNextTriggerIDKey gets the key for getting the next trigger ID.
+func GetNextTriggerIDKey() []byte {
+	return NextTriggerIDKey
+}
+
+// GetQueueKeyPrefix gets the queue's key prefix
+func GetQueueKeyPrefix() []byte {
+	return QueueKeyPrefix
+}
+
+// GetQueueKey gets the key for storing in the queue.
+func GetQueueKey(index uint64) []byte {
+	indexBytes := GetQueueIndexBytes(index)
+
+	key := QueueKeyPrefix
+	key = append(key, indexBytes...)
+	return key
+}
+
+// GetQueueStartIndexKey gets the key for storing the start index
+func GetQueueStartIndexKey() []byte {
+	return QueueStartIndexKey
+}
+
+// GetQueueLengthKey gets the key for storing the queue length
+func GetQueueLengthKey() []byte {
+	return QueueLengthKey
+}
+
+// GetQueueIndexFromBytes returns the index in uint64 format from a byte array
+func GetQueueIndexFromBytes(bz []byte) (start uint64) {
+	return binary.BigEndian.Uint64(bz)
+}
+
+// GetQueueIndexBytes returns the byte representation of the queue index
+func GetQueueIndexBytes(index uint64) (queueIndexBz []byte) {
+	queueIndexBz = make([]byte, QueueIndexLength)
+	binary.BigEndian.PutUint64(queueIndexBz, index)
+	return
+}
+
+// GetTriggerIDFromBytes returns triggerID in uint64 format from a byte array
+func GetTriggerIDFromBytes(bz []byte) (triggerID TriggerID) {
+	return binary.BigEndian.Uint64(bz)
+}
+
+// GetTriggerIDBytes returns the byte representation of the trigger ID
+func GetTriggerIDBytes(triggerID TriggerID) (triggerIDBz []byte) {
+	triggerIDBz = make([]byte, TriggerIDLength)
+	binary.BigEndian.PutUint64(triggerIDBz, triggerID)
+	return
+}
```

Comment:
While I like named return variables for a lot of things, they don't always need to be named. When a function has a named return variable, that variable is created right at the start of the function execution. Then, if the `return` has values, those are copied into the named return variables before calling any of the deferrables and ultimately returning the values to he caller.

In the case of `GetTriggerIDFromBytes`, you're basically needlessly creating the `triggerID` variable.

Also, even though I'm okay with naked `return`s (like in `GetTriggerIDBytes`), some folks are really uncomfortable with them. Since `GetTriggerIDBytes` has the named return variable, and that's what's used in the func, it's better to have the naked return than, e.g. `return triggerIDBz`. Doing that can cause needless copying.

In the cases in this file, I quite like using named return variables, specially for the funcs that return a `[]byte`.